File: Leetcode/linkedlist/83_RemoveDuplicateFromLinkedList/Solution.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
    def deleteDuplicates(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """
        if not head: return head
        temp = head;
        while temp:
            while temp.next and temp.val==temp.next.val:
                temp.next = temp.next.next
            temp = temp.next
        return head

File: Leetcode/linkedlist/83_RemoveDuplicateFromLinkedList/Solution.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode* deleteDuplicates(ListNode* head) {
        if(head==nullptr) return head;
        ListNode* temp = head;
        while(temp!=nullptr){
            while(temp->next!=nullptr && temp->val==temp->next->val){
                temp->next = temp->next->next;
            }
            temp = temp->next;
        }
        return head;
    }
};